include/poller.php

```diff
@@ -19,15 +19,17 @@
 require_once("boot.php");
 
 function poller_run($argv, $argc){
-	global $a, $db, $poller_up_start, $poller_db_duration;
+	global $a, $poller_up_start, $poller_db_duration;
 
 	$poller_up_start = microtime(true);
 
-	$a = new App(dirname(__DIR__));
+	if (empty($a)) {
+		$a = new App(dirname(__DIR__));
+	}
 
-	@include(".htconfig.php");
-	require_once("include/dba.php");
-	$db = new dba($db_host, $db_user, $db_pass, $db_data);
+	require_once ".htconfig.php";
```

Comment:
Good, you found this. But generally this will require PHP to scan the whole include path. If you use `strace php index.php` you will see a lot `lstat` system calls. You can avoid them by using absolute paths (I call that Full-Qualified File Name, FQFN in short). I would recommend here to write a helper method in `App` class, like `include_script()` and `require_script()` which then uses the in constructor of `App` provided path for FQFN-generation.

Reply:
This would be some additional feature and not part of this work.

Reply:
This is marginal low-level optimization, let's keep that for later.

Reply:
Okay. 👍